### ISSUE TITLE
Intepretation of Persistence Curve Results

### ISSUE THREAD

**Issue by hollowsunhc** [topology-tool-kit/ttk#1009]
Hi! This post is a discussion rather than a bug report/feature request. I wonder if it's possible to enable discussions in this project!

I've been trying to make sense of the results given by the Persistence Curve on a minimal example.

I generated a square plane centered at origin, of length $2\pi$ and applied a calculator with the following function: `cos(coordsX)+cos(coordsY)`

I get the following:

![image](https://github.com/topology-tool-kit/ttk/assets/128929460/ae652f36-c88a-47af-8126-f06e971cb8bd)

where cyan numbers are values of the function.

I'm using TTK 1.0.0 at the moment so I have to apply the Persistence Curve directly onto the Calculator. The following `Minimum-saddle pairs` table is then presented to me:

| Row ID | Persistence | Number Of Pairs |
|--------|-------------|-----------------|
| 0      | 2           | 4               |
| 1      | 2           | 3               |
| 2      | 2           | 2               |
| 3      | 4           | 1               |

I either have no clue how to interpret the last row or there's something fishy going on with my fork maybe... Is this normal?

Thank you!
Patrick Laurin

**Comment by julien-tierny** [comment]
Hi Patrick,
thanks a lot for your input.
What happens in that case is the following:
- you have 3 persistence pairs, representing 3 connected components created in three distinct minima (at value -2), which each merge with a pre-existing component at a saddle point (at value 0). these three pairs have a persistence of (0 - (-2) = 2).
- then you have one class with infinite persistence, representing the connected component of sublevel set created at the global minimum of the dataset (chosen "arbitrarily" among the 4 minima). this component never merges with a pre-existing one. then it has infinite persistence. in ttk, by convention, for the corresponding pair in the diagram, we set the cellArray "isFinite" to 0 and we set its persistence to its lifespan in the function range (i.e. globalMax - pairBirth = 2 - (-2) = 4).

so in short, everything looks OK. maybe the encoding of the class with infinite persistence was a little confusing (but correct). thanks for letting us know if you need any further information.

best,

**Comment by hollowsunhc** [comment]
Hi Julien,

Thank you for the clarification! If my understanding is correct, one of the minima is designated for the computation of infinite persistence, leaving the other three minima to form distinct connected components that merge at saddle points. This explanation aligns with why the maximum-saddle pair is associated with the computation of infinite persistence

| Row ID | Persistence (maximum-saddle) | Number Of Pairs (maximum-saddle) |
|--------|-------------|-----------------|
| 0      | 4           | 1               |

because I only have one pair which is also used for the computation of the infinite persistence.

Consequently, I now understand how the "All Pairs" table aggregates the results from three categories: maximum-saddle, minimum-saddle, and saddle-saddle pairs.

Thank you!
Patrick Laurin
